src/lib/features/integration-events/integration-events-service.ts

```diff
@@ -0,0 +1,57 @@
+import type { Logger } from '../../logger';
+import type { IFlagResolver, IUnleashConfig } from '../../types';
+import type {
+    IntegrationEventsStore,
+    IntegrationEventWriteModel,
+} from './integration-events-store';
+import type { IntegrationEventSchema } from '../../openapi/spec/integration-event-schema';
+
+export class IntegrationEventsService {
+    private readonly logger: Logger;
+    private integrationEventsStore: IntegrationEventsStore;
+    private flagResolver: IFlagResolver;
+
+    constructor(
+        {
+            integrationEventsStore,
+        }: { integrationEventsStore: IntegrationEventsStore },
+        {
+            getLogger,
+            flagResolver,
+        }: Pick<IUnleashConfig, 'getLogger' | 'flagResolver'>,
+    ) {
+        this.integrationEventsStore = integrationEventsStore;
+        this.flagResolver = flagResolver;
+        this.logger = getLogger('integration-events-service');
+    }
+
+    isEnabled(): boolean {
```

Comment:
Entirely possible this is purely personal taste but I would leave abstractions off flag resolutions. Abstractions are most useful when they're hooks for future change but flags should only change to go away in the very near future. Won't complain if you want to keep it though

Reply:
Agreed, this is unnecessary and potentially even dangerous. Addressed in https://github.com/Unleash/unleash/commit/278f7c7fe2091607f130cc95386a5f167166ee69